### DISCUSSION TITLE
My commits do not get added to my contributions

### DISCUSSION THREAD

**Discussion by MrQui3** [community/community#68849] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

![image](https://github.com/community/community/assets/51865142/a2e98f69-4223-4bd5-a27e-a51f2c963abe)
![image](https://github.com/community/community/assets/51865142/222db3ab-7cb4-47eb-8d20-180c093bc851)

"Only new repositories are displayed in my contributions. However, when I make commits, they are not shown there, whether the repository is public or private. Or do just verfied commits count.
Edit: This problem is in all of Projects and they aren't forked or smth like that.

**Comment by AnkurBansal18** [comment]
Hi @MrQui3, If your commits are not showing up in the GitHub contribution graph (often referred to as the "GitHub heatmap" or "GitHub activity graph"), there could be several reasons for this issue. Here are some common factors to consider:

Visibility Settings: Ensure that the repository where you are making commits is set to "Public" or "Private" based on your requirements. If it's a private repository, be aware that your contributions may not appear in your public profile.

Email Address: GitHub associates commits with your GitHub account based on the email address used in the commit. Make sure the email address you used for your Git commits matches the email address associated with your GitHub account.

To check the email address in your Git configuration, run: git config user.email
To set the email address in your Git configuration, run: git config --global user.email "your.email@example.com"

Commit Authorship: Ensure that your commits have the correct authorship information, including your GitHub username and the email address associated with your GitHub account. You can set this information in your Git configuration:

To check the username in your Git configuration, run: **git config user.name**
To set the username in your Git configuration, run: **git config --global user.name "Your GitHub Username"**

**Email Privacy Settings:** Check your email privacy settings on GitHub. If your email privacy settings are configured to keep your email address private, your contributions may not appear in the graph. To check and modify your email privacy settings:

Go to your GitHub profile settings.
Navigate to "Emails" under "Account settings."
Ensure that the email address you use for commits is verified and set to "Public."

Commit Time: The GitHub contribution graph may take some time to update. If you've recently made commits, it may take a while for them to appear in the graph.

Repository Visibility: Make sure you are committing to a repository that is associated with your GitHub account. Contributions to repositories owned by organizations may not always appear on your personal contribution graph.

Repository Forks: If you forked a repository and are making commits to your fork, those contributions may not appear on the original repository's contribution graph. They will be visible on your fork's contribution graph. (might not the case with you as your projects aren't forked).

Cached Data: Occasionally, GitHub's contribution graph may experience caching issues. If you believe you should have contributions that are not showing up, you can try clearing your browser cache or waiting for some time for the graph to update.

User Agent or Browser Extensions: Some browser extensions or custom user agent settings can affect how GitHub loads and displays data. Try accessing your GitHub profile on a different browser or in incognito/private mode to rule out any issues related to browser settings or extensions.

if still it doesn't work, try contacting github support.

**Reply by MrQui3** [reply]
Hi AnkurBansal18, thanks for your answer. I did already everything you said. My github profile is connected with my pycharm account and I did every commit over pycharm. In github the commits are shown, but they are not included in the github heatmap(thanks for the name). Any idea why. This problem is in every repository I have, so it has nothing to do with a fork or something like that.
How do I contact github support

**Reply by radar-blip** [reply]
[Support](https://support.github.com/request)

**Reply by AnkurBansal18** [reply]
Hey, If your commits are not showing up in the GitHub heatmap, even though you can see the commits in the repository's commit history, you can try this last possible solution also:

The GitHub heatmap is based on the **timestamps** associated with your commits. Ensure that the timestamps of your commits are accurate and match your local time zone settings(check your device's settings). If your timestamps are significantly different from your actual commit times or are in a different time zone, it can affect how they appear on the heatmap.

To check the timestamp of a commit, you can use the following Git command:

**git show --format=fuller --date=iso <commit-SHA>**

_Replace <commit-SHA> with the SHA-1 hash of the commit you want to inspect_.


If this also doesn't work, then you can create a ticket using this, and GitHub support will look into the issue.
[https://docs.github.com/en/support/contacting-github-support/creating-a-support-ticket](url)

I hope it helps :)

**Reply by 0xTanzim** [reply]
The information provided is highly informative and valuable.

**Reply by rainyPoseidon21** [reply]
@MrQui3, as you can see that your commit messages author are MrQui rather than MrQui3. So I figured that is what u added under "Local" git config (the one under your project directory). To check this, you can use the same command to check the global git username/ email, but without --global. In order for your git to override local config, and start using global ones, u need to remove the user details from the local .gitconfig file. To do this, you need to change the hidden file visibility from your machine, navigate to that local .gitconfig, remove the user part, along with its subcomponents.

**Reply by MrQui3** [reply]
oh i already solved it a long time ago

**Reply by rishantenis** [reply]
Could you tell us how you solved that issue?

**Reply by Orel6505** [reply]
> Could you tell us how you solved that issue?

On possible solution is to set the branch you uploaded as main branch

**Reply by xdlemar** [reply]
nice

**Reply by thestonegod** [reply]
I hope 1 of these solves my problem.

**Reply by tarnzolo123-cell** [reply]
Hi

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @MrQui3 ! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the GitHub product Discussions. I've gone ahead and moved it for you. Good luck!

**Comment by balwierz** [comment]
I have the same issue on my public repository. Commits from the web (firefox or chrome on Android) did not end up counted in my heatmap. While posting them I received an error (file can not be created) although the commit happened and the file was created. 
See the heatmap on https://github.com/balwierz: it says "no commits on 2023-10-07", but there is a commit: https://github.com/balwierz/LeetCode/commit/cbd536836c94080282e637cdd5047996ea83f200 and the repo status is "Public"

**Comment by luisgvecchio** [comment]
Hi! Have the same issue. Did you manage to fix this? thanks!

**Comment by roman-sundurov** [comment]
I have the same issue.

**Comment by faridnec** [comment, marked as answer]
did you check the globally configured email address (across all repositories on your machine)?
```bash
git config --global user.email
```
make sure that the displayed email address matches the one associated with your GitHub account. If they don't match, you can update the email address in your Git configuration by
```bash
git config --global user.email "your.email@blabla.com"
```
hope this helps

**Reply by roman-sundurov** [reply]
Yes, I checked global configured email on mac, in commits and on github. They are identical.

**Reply by roman-sundurov** [reply]
@faridnec By the way, I usually use the GitKraken client to create new branches and commits.

**Reply by faridnec** [reply]
hmm, I am not sure, maybe try creating a new commit using the command line with the correct email address and see if it reflects correctly on GitHub. This can help isolate whether the issue is specific to GitKraken or if there's a broader configuration problem.

**Reply by jamilwel** [reply]
git config --global user.email

**Reply by jamilwel** [reply]
[billing_data_20251016.csv](https://github.com/user-attachments/files/22938806/billing_data_20251016.csv)

**Comment by AntoBongio** [comment]
I have the same issue, any new advise?

**Reply by 0xTanzim** [reply]
I noticed that when I commit in another branch, it is not added. Did you commit to the main branch?

**Reply by AntoBongio** [reply]
Nope, I commit on a different branch. Do you think that it could be correlated?

**Reply by 0xTanzim** [reply]
I have noticed that when I commit to a different branch, it is not being added.

**Reply by Orel6505** [reply]
Same problem happens to me

**Reply by saramasarone** [reply]
Same problem here - I've checked everything that is being mentioned here

**Reply by euncheoly** [reply]
it seems like the commits on branches other than the default main branch get counted for contribution after they are merged into and pushed to the default main branch.

**Reply by nameerakhter** [reply]
> Nope, I commit on a different branch. Do you think that it could be correlated?

It will only display the contributions on the main branch so if your branch gets merged to main or you directly push commits to main in both cases the commits will show up in your heatmap

**Comment by johnwaweru** [comment]
I had a simple case of a misspelled email and had to update all my commits per repo as follows:

```
git filter-branch --env-filter 'if [ "$GIT_COMMITTER_EMAIL" = "misspelled_email@example.com" ]; then
    export GIT_COMMITTER_EMAIL="corrected_email@example.com";
fi;
if [ "$GIT_AUTHOR_EMAIL" = "misspelled_email@example.com" ]; then
    export GIT_AUTHOR_EMAIL="corrected_email@example.com";
fi' -- --all
```

Once done, you will need to run:

`git push --force --all`

Your heat map should now be updated.

**Reply by dav15jr** [reply]
Thanks OR Asante Sana 🙌

**Comment by nameerakhter** [comment]
Hey recently I am also having the same issue I have checked everything I am on the main branch the commits are done by me the github heatmap settings says that I have public and private contributions both turned on in preferences still they are not showing up it was working until yesterday
![image](https://github.com/user-attachments/assets/32db70a8-ab8d-4d1d-a48b-ae1b98f42556)
![image](https://github.com/user-attachments/assets/c8509555-8fec-436a-b9f2-fbd28b4a57f8)

**Reply by skartik-sk** [reply]
Same issue. Does anyone found any new solution?

**Reply by nameerakhter** [reply]
I think there is a massive delay right now the commits are showing up after 2-3 hours after being pushed mine just started showing up but it is very weird I also experienced it for the first time.

**Reply by skartik-sk** [reply]
yeah Same just got updated.

**Reply by EliR94** [reply]
Same issue, mine also is not showing from yesterday and today. I checked a few times and not updated! They were working fine up until this point. They are commits in a private repo but my settings include private and has been working for public and private up until now, very strange

**Reply by nameerakhter** [reply]
For me it took almost upto a day for the previous ones to show up but now it's working fine maybe wait till tomorrow if the problem still persists try raising a ticket on github support

**Comment by amir2628** [comment]
So if I am a developer and I am working at a company, we set up a repository which logically will be a private one, and I am added as a contributor, my contributions will not be reflected on my GitHub heatmap?

My email address and user account in terminal, are the same as the ones in my GitHub account. I am pushing changes in a branch of a private repository, and my contributions do not show in the heatmap.
So I am working for months on this private repository, and do not have time to work on anything else, but the heatmap does not show my efforts. 😄 

Is there a reason for this? I think at least the green indicators should appear so that people do not assume my account is dead... 🚬

**Comment by Mohamezezo** [comment]
How can i get cash out

**Reply by 09375876590ka-droid** [reply]
سلام وقت بخیر چگونه  میتوانم پول نقد دریافت  کنم

**Comment by justpayme3** [comment]
Spam

**Comment by vladyslavk0** [comment]
I have the same issue, any new advise?

**Comment by 5515W90438** [comment]
Gracias por su atención

**Comment by Kamil777** [comment]
Nie nie mam niestety, zabawa się skonczy)ła

wt., 15 lip 2025, 14:06 użytkownik El-qade ***@***.***>
napisał:

**Comment by 670698-rgb** [comment]
>

**Comment by 670698-rgb** [comment]
ghujkd,

**Comment by chzoloo39-bit** [comment]
[Судалгаа.xlsx](https://github.com/user-attachments/files/22876056/default.xlsx)

**Comment by KotChernyj** [comment]
> ### Select Topic Area
> 
> Question
> 
> ![image](https://github.com/community/community/assets/51865142/a2e98f69-4223-4bd5-a27e-a51f2c963abe)
> ![image](https://github.com/community/community/assets/51865142/222db3ab-7cb4-47eb-8d20-180c093bc851)
> 
> 
> "Only new repositories are displayed in my contributions. However, when I make commits, they are not shown there, whether the repository is public or private. Or do just verfied commits count.
> Edit: This problem is in all of Projects and they aren't forked or smth like that.
>

**Reply by KotChernyj** [reply]
Ja

**Reply by KotChernyj** [reply]
Ja

**Comment by BIGPHOR** [comment]
Code

**Comment by tarnzolo123-cell** [comment]
Good boy

**Comment by tarnzolo123-cell** [comment]
I love my mother so much

**Comment by raquelpn1979-tech** [comment]
https://docs.github.com/en/enterprise-server@3.17/pull-requests/how-tos/commit-changes/managing-branches-within-your-repository

**Comment by afayz4684-star** [comment]
> ### Select Topic Area
> 
> Question
> 
> ![image](https://github.com/community/community/assets/51865142/a2e98f69-4223-4bd5-a27e-a51f2c963abe)
> ![image](https://github.com/community/community/assets/51865142/222db3ab-7cb4-47eb-8d20-180c093bc851)
> 
> 
> "Only new repositories are displayed in my contributions. However, when I make commits, they are not shown there, whether the repository is public or private. Or do just verfied commits count.
> Edit: This problem is in all of Projects and they aren't forked or smth like that.
> 

- - - [ ]

**Comment by Edgarruiz8585** [comment]
> ### Select Topic Area
> 
> Question
> 
> ![image](https://github.com/community/community/assets/51865142/a2e98f69-4223-4bd5-a27e-a51f2c963abe)
>{Edgar_Manuel_Ruiz_Arias_217596670, title={Edgar Manuel Ruiz Arias mr ed}, journal= 279256497+Edgarruiz8585@users.noreply.github.com![image](https://github.com/community/community/assets/51865142/222db3ab-7cb4-47eb-8d20-180c093bc851)
> 
> 
> "Only new repositories are displayed in my contributions. However, when I make commits, they are not shown there, whether the repository is public or private. Or do just verfied commits count.
> Edit: This problem is in all of Projects and they aren't forked or smth like that.
>
